### ISSUE TITLE
feat(pr-5): survae_flow + InverseFlow + ELBO training

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#12]
## Problem / Request

Factor SurVAE convenience into one entry point and extend training to handle ELBO-mode flows: a `survae_flow` factory, an `InverseFlow` wrapper for variational posteriors, and ELBO training support in the training loop. `SurVAEFlow` itself already exists (`src/gauss_flows/_src/flows/survae.py`, with `log_prob(x, key, condition)` / `sample(key, sample_shape, condition)`); this issue is the ergonomic layer and the trainer.

## User Story

> As a user composing bijections, surjections, and stochastic transforms, I want a one-call constructor and a trainer that handles the ELBO case, so that mixed chains train as easily as pure-bijection flows do today.

## Motivation

Pure-bijection flows have `gaussianization_flow` + `fit_gaussianization_flow`; mixed SurVAE chains currently require hand-assembling `SurVAEFlow` and hand-writing a keyed training loop. This is the last blocking feature for the Phase-2 example notebooks (#16, #17, #18).

## Proposed API

Extend `src/gauss_flows/_src/flows/` (factory) and `src/gauss_flows/_src/inference/train.py` (trainer):

```python
def survae_flow(key, *, n_dims, transforms, base_dist=None) -> SurVAEFlow:
    """Thin constructor; defaults base_dist = Normal(jnp.zeros(n_dims))
    (or tuple event shapes for images)."""

class InverseFlow(eqx.Module):
    """Generative direction is forward; used for variational posteriors."""
    def sample_with_log_prob(self, key, n_samples) -> tuple[Array, Array]:
        ...  # shapes: ((n_samples,) + event_shape, (n_samples,))
```

- `fit_gaussianization_flow` detects `SurVAEFlow` and trains via ELBO (`−log_prob(x, key)`).
- **Investigation item:** check whether `flowjax.train.fit_to_data` can thread a `key` argument through the loss; if not, write a minimal optax + equinox training loop for the `SurVAEFlow` case (same early-stopping/patience semantics as the bijection path).

## References & Existing Code

- `SurVAEFlow`: `src/gauss_flows/_src/flows/survae.py` (already vmaps `log_prob`/`sample` over leading `sample_shape`)
- Surjections/stochastic transforms: `Slice`, `Augment`, `VAE` (`src/gauss_flows/_src/transforms/surjections/`, `.../stochastic/vae.py`)
- Trainer to extend: `src/gauss_flows/_src/inference/train.py`
- Prior refs (all landed): #7 (AbstractSurjection + SurVAEFlow), #8 (simple surjections), #9 (Slice/Augment), #10, #11 (VAE), #13 (marginal stubs)

## Implementation Steps

- [ ] Add `survae_flow` factory (module: `src/gauss_flows/_src/flows/survae.py` or a sibling)
- [ ] Add `InverseFlow` with `sample_with_log_prob`
- [ ] Extend `fit_gaussianization_flow` (or add a dispatch path) for keyed ELBO training
- [ ] Export `survae_flow`, `InverseFlow` via `src/gauss_flows/__init__.py`

## Acceptance Criteria

- [ ] `survae_flow` accepts any mix of bijections / surjections / stochastic transforms
- [ ] Training converges for: (a) bijection-only chain (regression vs `gaussianization_flow`), (b) chain with `Slice` (exact ELBO), (c) chain with `VAE` (lower-bound ELBO)
- [ ] `InverseFlow.sample_with_log_prob` returns shape `((n_samples,) + event_shape, (n_samples,))`
- [ ] Determinism under fixed key

## Definition of Done

- [ ] Code lands at the intended paths; public API exported
- [ ] Tests pass: `make test` (new `tests/test_survae_flow.py`, `tests/test_inverse_flow.py`)
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Docstrings (Google-style, **Shape:** + **Example:**) on all public symbols

## Testing

- [ ] Unit: factory defaults, export surface, `InverseFlow` shapes
- [ ] Regression: bijection-only `survae_flow` matches `gaussianization_flow` NLL
- [ ] Integration: ELBO decreases on `Slice` and `VAE` chains (small toys)

## Relationships

- Parent (wave epic): #20
- Blocked by: — (all prerequisite PRs landed)
- Blocks: #16, #17, #18 (VAE / multiscale / PriorCVAE notebooks)

**Comment by jejjohnson** [comment]
**normflows reference** (Composite / Reverse mapping):

- [`normflows/flows/base.py::Composite`](https://github.com/VincentStimper/normalizing-flows/blob/master/normflows/flows/base.py) ≈ our `SurVAEFlow` container (PR-0) — direct concept match.
- [`normflows/flows/base.py::Reverse`](https://github.com/VincentStimper/normalizing-flows/blob/master/normflows/flows/base.py) ≈ our `InverseFlow` — swaps forward/inverse semantics.

No new design decisions here; just noting the naming convergence. FlowJax already has `Invert` which is used today in `gaussianization_flow`; `InverseFlow` in PR-5 is the SurVAE-aware variant (supports stochastic layers with key threading).
